### ISSUE TITLE
Add .htaccess like routing support

### ISSUE THREAD

**Issue by csharrison** [web-platform-tests/wpt#14707]
Some URLs the browser requests (like those pointing to .well-known addresses) are fixed and cannot be modified by the web platform (without adding test only hooks).

It would be good if those URLs could return dynamically generated content without pointing to a .py file. This could be easily done via some sort of URL routing mechanism like htaccess.

**Comment by Hexcles** [comment]
`wptserve` has a router interface: https://wptserve.readthedocs.io/en/latest/router.html But it seems like this is not intended to be used by individual custom handlers?

**Comment by csharrison** [comment]
For special cases, is it OK practice to just add one-offs to RoutesBuilder.add_mount_point? It seems like there are a few special routes there for service workers, etc.

**Comment by gsnedders** [comment]
How many things do we have where we need things like `.well-known` addresses returning different responses on a per test file granularity? That's very hard to achieve when we might have multiple tests being run at once from a single server.

**Comment by jgraham** [comment]
Yeah, generally stuff that adds global state is bad, and the url routing table is global state. So if one test wants to do something funky with .well-known, all the tests accessing it have to know what that thing is to allow for it. So having an easy mechanism to do this is probably bad, but maybe for cases where it's really necessary adding to the routes is probably fine.

**Comment by csharrison** [comment]
An existing .well-known address that might plausibly want this is the existing origin-policy files in .well-known/origin-policy, if they wanted to point to different origins rather than "self" like web-platform-test:<port>.

If the port can be hardcoded, or templated in (without adding .sub), that would also work.

**Comment by foolip** [comment]
@csharrison is this something you'd be interested in submitting a PR for? If so can I assign the issue to you? (Trying to have assignees for all priority:roadmap issues.)

**Comment by csharrison** [comment]
I am happy to own this, but I'm not sure there's consensus that this is something we even want to ship, especially if it would end up being very complicated. Maybe there are alternatives we should explore instead?

My requirement is to allow dynamic content / templating on files with fixed paths (.well-known addresses). The only reason I need this is to construct valid origins, since ports are not known apriori. If there's some other mechanism to do that that's less general than .htaccess style routing, I'd be happy with that.

A simpler idea would be a separate whitelist of directories/files where .sub is implied by default, so they go through templating without needing to change file names.

**Comment by foolip** [comment]
@csharrison it seems indeed like something simpler might be possible: hard-wire a certain path in wptserve to be handled by a specific file, with comments in both places explaining what's going on. If we gain a second and a third case of the same, then we can generalize.
